MAGECLOUD-3667 New Relic Infrastructure (#4621)
magento/devdocs: guides/v2.1/cloud/project/new-relic.md

---
group: cloud-guide
title: New Relic services
functional_areas:
  - Cloud
  - Integration
  - Setup
  - Services
---

> [New Relic Infrastructure](https://newrelic.com/products/infrastructure) automatically connects with your application data and performance analytics to provide dynamic server monitoring and is standard for {{site.data.var.ece}} accounts. [New Relic for application performance management (APM)](https://docs.newrelic.com/docs/apm/new-relic-apm/getting-started/introduction-new-relic-apm) is a software analytics product that helps you analyze and improve application interactions. New Relic APM provides the following features to Magento:

Reviewer: Noticed APM in the last sentence. Should it be removed?

Author: Heya! <wave>
Actually, APM is still a thing. It is just not the ONLY thing. :) I might receive MORE information about this soon, so we shall see!

Reviewer 2: Do we need to mention that Infra is for Pro Production only?

Reviewer 3: I've seen Infrastructure available on Pro staging as well.

Reviewer 2: As discussed in Slack - that looks to be a provisioning bug. Working with infra team remove it from all stagings if it's there.

-  **Focus on specific transactions**—Actively mark and monitor key customer actions in your site, such as adding to the cart, checking out, or processing a payment.
-  **Database query monitoring**—Locate and monitor database queries affecting performance.
-  **App Map**—View all application dependencies within your Magento site, extensions, and external services.
-  **Apdex scores**—Evaluate performance and create alerts that identify issues and notify you when they occur, such as site performance affected by a flash sale or web event.

{:.bs-callout .bs-callout-info}

> Your {{site.data.var.ece}} account includes the software for the New Relic service along with a license key. You do not need to purchase or install any additional software. For Pro accounts, if New Relic is not installed on the Staging and Production environments, submit a [support ticket]({{ page.baseurl }}/cloud/trouble/trouble.html) to request installation.

Reviewer 2: For PRO only APM goes to STG+PROD. Infra goes to PROD only. This needs to be clear here.

1.  Ask the new user to check their email for a New Relic notification with account information.

See [Accounts](https://docs.newrelic.com/docs/accounts/accounts) in the New Relic documentation for information about user roles, managing user accounts, and changing account ownership.

## Configure New Relic

Pro accounts are preconfigured to use the New Relic service; however, Starter accounts must add the New Relic service to the `.magento.app.yaml` configuration file and assign the license key to an environment variable. You can apply the New Relic license key to a maximum of three active environments.

For Starter projects, check if the `.magento.app.yaml` file includes the New Relic extension in the `runtime` list. If not, then add the following:

> .magento.app.yaml

```yaml
runtime:
    extensions:
        - newrelic
```

## New Relic license key

> The {{site.data.var.ece}} Starter plans support up to three instances of your New Relic license across all environments. We recommend adding a New Relic license key to your Pro Staging and Production environments, and you can add the license to one other environment of your choice. 

Reviewer 2: There is a mention of PRO here - is that a typo?

Author: Yes, a total typo. removed it.

{:.bs-callout .bs-callout-info}

> If you have more than three active environments using the same New Relic license key, you need to remove a license variable from an existing environment. Contact your Magento technical account manager or New Relic sales (sales@newrelic.com) if you want to use New Relic on more than three environments.

Reviewer 2: I assume we need to remove sales from here - we can connect them once required with our TAM on NR side.

You can also access the license key from the Magento Cloud Project interface. See [Blackfire and New Relic credentials]({{ page.baseurl }}/cloud/project/projects.html).

### Add a license key to an environment

For Starter accounts, you need to use the license key and set a variable in the environment.

1.  Change to the environment that needs the license key.

1.  Set the variable.

    ```bash
    magento-cloud variable:set php:newrelic.license <newrelic-license-key>
    ```

Adding a variable to an environment triggers a redeployment to update the environment variables and values. See [Working with variables]({{page.baseurl}}/cloud/env/working-with-variables.html).

### Remove a license key from an environment

1.  List environment variables.

    ```bash
    magento-cloud vget
    ```

    Response:

    ```terminal
    +----------------------+-------------+------------------------+
    | Name                 | Level       | Value                  |
    +----------------------+-------------+------------------------+
    | php:newrelic.license | environment | <newrelic-license-key> |
    +----------------------+-------------+------------------------+
    ```
    {: .no-copy}

> 1.  Delete a variable.

Reviewer: Does removing the variable also trigger redeploy?

Author: Why, yes it does! 🧐

1.  Ask the new user to check their email for a New Relic notification with account information.

See [Accounts](https://docs.newrelic.com/docs/accounts/accounts) in the New Relic documentation for information about user roles, managing user accounts, and changing account ownership.

## Configure New Relic

> Pro accounts are preconfigured to use the New Relic service; however, Starter accounts must add the New Relic service to the `.magento.app.yaml` configuration file and assign the license key to an environment variable. You can apply the New Relic license key to a maximum of three active environments.

Reviewer 4: So for Pro accounts, you don't need to configure anything. Do you need to manage license keys for Pro environments?

If not, it might be clearer to make a heading:  " Add New Relic extension to your Starter project, and make it a level 3 heading.  

Same with the add a license key to an environment, remove license key -- if the instructions apply only to Starter projects.  Or find another way to clarify that you only need to do these things for Starter plan projects.

Author: This is already done for **new** starter projects, but may not be the case in all Starter projects. This configuration step is more of a verification.

I will not make an h3 in this case because the key is not part of the yaml configuration. 

Any Cloud project can manipulate variables in a paas environment.

Author: Your comments here confuse me. I have a sentence in the license key instructions that says "For Starter accounts, you need to use the license key and set a variable in the environment." I'm not sure why that is unclear. In fact, even if you are a Pro customer, you can do these steps. You probably don't need to, which is why I list how to find and verify things.

What I can do, is add a little more about verifying. I will think about it more.

Reviewer 4: The main clarification I was looking for is that for Pro environments, you don't need to do the configuration since it's already done for you.  Since you can configure New Relic for integration branches on Starter, then it's likely that you might need to change the configuration if your integration environments change.  But on Pro (if I understand correctly), you only have New Relic on Staging and Production, and those branches are stable and not likely to change. 

On Pro, the keys are set up for you during provisioning?

1.  Ask the new user to check their email for a New Relic notification with account information.

See [Accounts](https://docs.newrelic.com/docs/accounts/accounts) in the New Relic documentation for information about user roles, managing user accounts, and changing account ownership.

## Configure New Relic

> Pro accounts are preconfigured to use the New Relic service; however, Starter accounts need to verify that the New Relic service is configured properly. For Starter projects, check if the `.magento.app.yaml` file includes the New Relic extension in the `runtime` list. If not, then add the following:

Reviewer 4: ```suggestion
Pro accounts are preconfigured to use the New Relic service; however, customers with Starter accounts need to verify that the New Relic service is configured properly. For Starter projects, check if the `.magento.app.yaml` file includes the New Relic extension in the `runtime` list. If not, then add the following:
```

All collected data details the time spent on an action transmitting data, queries, or _Redis_ data. If queries cause issues, New Relic provides information to track and respond to it.

To learn more about using and customizing New Relic services for your project, see [New Relic Infrastructure documentation](https://docs.newrelic.com/docs/infrastructure/new-relic-infrastructure/getting-started/introduction-new-relic-infrastructure)

## New Relic and Blackfire

> You can use the high-level data captured by New Relic APM to see what is happening in your application in realtime. The New Relic Insights helps to focus on events, transactions, regions, etcetera, to correlate data.  Also, with the code-level data captured by the [Blackfire Profiler]({{ page.baseurl }}/cloud/project/project-integrate-blackfire.html) you get a comprehensive view of your Magento store performance.

Reviewer 4: ```suggestion
You can use the high-level data captured by New Relic APM to see what is happening in your application in realtime. The analytics from New Relic Insights help to focus on events, transactions, regions, etcetera, to correlate data.  Also, with the code-level data captured by the [Blackfire Profiler]({{ page.baseurl }}/cloud/project/project-integrate-blackfire.html) you get a comprehensive view of your Magento store performance.
```

Author: Removed this statement. Insights is not supposed to be here.

All collected data details the time spent on an action transmitting data, queries, or _Redis_ data. If queries cause issues, New Relic provides information to track and respond to it.

To learn more about using and customizing New Relic services for your project, see [New Relic Infrastructure documentation](https://docs.newrelic.com/docs/infrastructure/new-relic-infrastructure/getting-started/introduction-new-relic-infrastructure)

### New Relic and Blackfire

> You can use the high-level data captured by New Relic APM to see what is happening in your application in realtime. Combine that with the code-level data captured by the [Blackfire Profiler]({{ page.baseurl }}/cloud/project/project-integrate-blackfire.html) and you get a comprehensive view of your Magento store performance.

Reviewer 4: ```suggestion
You can use the high-level data captured by New Relic APM to see what is happening in your application in realtime. Combine that with the code-level data captured by the [Blackfire Profiler]({{ page.baseurl }}/cloud/project/project-integrate-blackfire.html) to get a comprehensive view of your Magento store performance.
```
